src/lib/operators/get_table.cpp

```diff
@@ -12,84 +12,176 @@
+        continue;
+      }
+
+      const auto chunk = stored_table->get_chunk(chunk_id);
 
       if (!chunk || (chunk->get_cleanup_commit_id() &&
                      *chunk->get_cleanup_commit_id() <= transaction_context()->snapshot_commit_id())) {
-        temp_excluded_chunk_ids.emplace_back(chunk_id);
+        deleted_chunk_ids.emplace_back(chunk_id);
       }
     }
   }
 
-  if (temp_excluded_chunk_ids.empty()) {
-    return original_table;
+  /**
+   * Early out if no pruning of Chunks or Columns is necessary
+   */
+  if (deleted_chunk_ids.empty() && _pruned_chunk_ids.empty() && _pruned_column_ids.empty()) {
+    return stored_table;
   }
 
-  // We create a copy of the original table, but omit excluded chunks
-  std::sort(temp_excluded_chunk_ids.begin(), temp_excluded_chunk_ids.end());
-  temp_excluded_chunk_ids.erase(std::unique(temp_excluded_chunk_ids.begin(), temp_excluded_chunk_ids.end()),
-                                temp_excluded_chunk_ids.end());
+  // We cannot create a Table without columns - since Chunks rely on their first column to determine their row count
+  Assert(_pruned_column_ids.size() < stored_table->column_count(), "Cannot prune all columns from Table");
+  DebugAssert(std::all_of(_pruned_column_ids.begin(), _pruned_column_ids.end(),
+                          [&](const auto column_id) { return column_id < stored_table->column_count(); }),
+              "ColumnID out of range");
+
+  /**
+   * Build pruned TableColumnDefinitions of the output Table
+   */
+  auto pruned_column_definitions = TableColumnDefinitions{};
+  if (_pruned_column_ids.empty()) {
+    pruned_column_definitions = stored_table->column_definitions();
+  } else {
+    pruned_column_definitions =
+        TableColumnDefinitions{stored_table->column_definitions().size() - _pruned_column_ids.size()};
+
+    auto pruned_column_ids_iter = _pruned_column_ids.begin();
+    for (auto stored_column_id = ColumnID{0}, output_column_id = ColumnID{0};
+         stored_column_id < stored_table->column_count(); ++stored_column_id) {
+      if (pruned_column_ids_iter != _pruned_column_ids.end() && stored_column_id == *pruned_column_ids_iter) {
+        ++pruned_column_ids_iter;
+        continue;
+      }
 
-  auto output_chunks =
-      std::vector<std::shared_ptr<Chunk>>{original_table->chunk_count() - temp_excluded_chunk_ids.size()};
+      pruned_column_definitions[output_column_id] = stored_table->column_definitions()[stored_column_id];
+      ++output_column_id;
+    }
+  }
+
+  /**
+   * Build the output Table, omitting pruned Chunks and Columns as well as deleted Chunks
+   */
+  auto output_chunks = std::vector<std::shared_ptr<Chunk>>{stored_table->chunk_count() - _pruned_chunk_ids.size() -
+                                                           deleted_chunk_ids.size()};
+  auto output_chunks_iter = output_chunks.begin();
+
+  auto pruned_chunk_ids_iter = _pruned_chunk_ids.begin();
+  auto deleted_chunk_ids_iter = deleted_chunk_ids.begin();
```

Comment:
Should we continue to treat pruned and deleted chunks differently up to here?

Reply:
treated uniformly now (`excluded_chunk_ids`)